libraries/stdlib/samples/test/samples/text/strings.kt

```diff
@@ -517,6 +518,154 @@ class Strings {
         assertPrints(mixedColor, "brown&blue")
     }
 
+    @Sample
+    fun toPattern() {
+        val string = "Kotlin [1-9]+\\.[0-9]\\.[0-9]+"
+        val pattern = string.toPattern(Pattern.CASE_INSENSITIVE)
+        assertPrints(pattern.pattern(), string)
+        assertTrue(pattern.flags() == Pattern.CASE_INSENSITIVE)
+        assertTrue(pattern.matcher("Kotlin 2.1.255").matches())
+        assertTrue(pattern.matcher("kOtLiN 21.0.1").matches())
+        assertFalse(pattern.matcher("Java 21.0.1").matches())
+        assertFalse(pattern.matcher("Kotlin 2.0").matches())
+
+        assertFails { "[0-9".toPattern(Pattern.CASE_INSENSITIVE) } // the given regex is malformed
+    }
+
+    @Sample
+    fun encodeToByteArray() {
+        val str = "Kòtlin\u00a02.1.255"
+        val byteArray = str.encodeToByteArray()
+        assertPrints(byteArray.contentToString(), "[75, -61, -78, 116, 108, 105, 110, -62, -96, 50, 46, 49, 46, 50, 53, 53]")
+
+        val byteArrayWithoutFirstLetter = str.encodeToByteArray(startIndex = 1, endIndex = str.length)
+        assertPrints(byteArrayWithoutFirstLetter.contentToString(), "[-61, -78, 116, 108, 105, 110, -62, -96, 50, 46, 49, 46, 50, 53, 53]")
+
+        val byteArrayWithoutLastLetter = str.encodeToByteArray(startIndex = 0, endIndex = str.length - 1)
+        assertPrints(byteArrayWithoutLastLetter.contentToString(), "[75, -61, -78, 116, 108, 105, 110, -62, -96, 50, 46, 49, 46, 50, 53]")
+
+        val byteArrayWithoutNbsp = str.replace("\u00a0", " ").replace("ò", "o").encodeToByteArray()
+        assertPrints(byteArrayWithoutNbsp.contentToString(), "[75, 111, 116, 108, 105, 110, 32, 50, 46, 49, 46, 50, 53, 53]")
+    }
+
+    @Sample
+    fun `subString(startIndex)`() {
```

Comment:
Please use _regular_ (not requiring backticks) names for sample functions. It seems like Dokka can't handle them in `@sample` tag.

Names don't have to be brief and could be something like `substringFromStartIndex`, `substringByStandAndEndIndices`, etc.